ignite/metrics/entropy.py

```diff
@@ -0,0 +1,82 @@
+from typing import Sequence
+
+import torch
+import torch.nn.functional as F
+
+from ignite.exceptions import NotComputableError
+from ignite.metrics.metric import Metric, reinit__is_reduced, sync_all_reduce
+
+__all__ = ["Entropy"]
+
+
+class Entropy(Metric):
+    r"""Calculates the mean of `entropy <https://en.wikipedia.org/wiki/Entropy_(information_theory)>`_.
+
+    .. math:: H = \frac{1}{N} \sum_{i=1}^N \sum_{c=1}^C -p_{i,c} \log p_{i,c},
+       \quad p_{i,c} = \frac{\exp(z_{i,c})}{\sum_{c'=1}^C \exp(z_{i,c'})}
+
+    where :math:`p_{i,c}` is the prediction probability of :math:`i`-th data belonging to the class :math:`c`.
+
+    - ``update`` must receive output of the form ``(y_pred, y)`` while ``y`` is not used in this metric.
+    - ``y_pred`` is expected to be the unnormalized logits for each class.
```

Comment:
Do we have any limitations on the shape of the predictions ? For example, if y_pred is [B, C, H, W] as for image segmentation task, can we still compute the value ?

Reply:
It works with only [B, C] shape now, but supporting segmentation would be nice!
I'd like to modify the code to support [B, C, H, W].

Reply:
I modified the code to support [B, C, ...] shape.
For instance, [B, C, H, W] tensors are reshaped into [B\*H\*W, C] and then entropy is computed over B\*H\*W predictions.